app/assets/javascripts/govuk_publishing_components/components/global-banner.js

```diff
@@ -0,0 +1,68 @@
+//= require govuk_publishing_components/lib/cookie-functions
+/*
+  Global banner
+
+  Manages count of how many times a global banner has been seen
+  using cookies.
+*/
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function GlobalBanner ($module) {
+    this.$module = $module
+    this.GLOBAL_BANNER_SEEN_COOKIE = 'global_banner_seen'
+    this.alwaysOn = this.$module.getAttribute('data-global-banner-permanent') === 'true'
+    this.bannerVersion = parseInt(this.$module.getAttribute('data-banner-version'))
+  }
+
+  GlobalBanner.prototype.init = function () {
+    // if no cookie consent just show the banner
+    // if there is cookie consent...
+    //   if there's no cookie or the banner should always be shown, set the cookie and show the banner
```

Comment:
It may be worth directly mentioning the `global_banner_seen` cookie here instead of just `no cookie` / `a cookie`, especially to differentiate this sentence from the previous one about cookie consent. I initially thought `a cookie` was referring to the consent cookie.

Reply:
Good point, have reworded, thanks.